doc: Add three new glossaries (#3640)
projectacrn/acrn-hypervisor: doc/glossary.rst

   High GM
      See :term:`Hidden GM`

   Hybrid Mode

>       One of three operation modes which ACRN can support. The mixed mode 

Reviewer: ```suggestion
      One of three operation modes (hybrid, partition, sharing) that ACRN supports. 
```

>       of supporting both partitioned user VM and sharing user VMs

Reviewer: ```suggestion
In this mixed mode, physical hardware resources can be both partitioned to individual user VMs and shared across user VMs.
```

      the Project ACRN architecture, pass-through devices are owned by the
      foreground OS.

   Partition Mode

>       One of three operation modes which ACRN can support. Physical 

Reviewer: ```suggestion
      One of three operation modes (partition, sharing, hybrid) that ACRN supports. 
```

>       hardware resources are partitioned to individual user VM.

Reviewer: ```suggestion
      Physical hardware resources are partitioned to individual user VMs.
```

      access hardware resources directly by running native drivers and
      provides device sharing services to User VMs via the Device Model.

   Sharing Mode

>       One of three operation modes which ACRN can support. Most of physical 

Reviewer: ```suggestion
      One of three operation modes (sharing, hybrid, partition) that ACRN supports. Most of the physical 
```
